src/main/java/com/treasuredata/client/TDClient.java

```diff
@@ -660,24 +660,41 @@ protected String toJson(Object any)
     }
 
     @Override
-    public TDSavedQuery saveQuery(TDSaveQueryRequest query)
+    public TDSavedQuery saveQuery(TDSaveQueryRequest request)
     {
-        String json = toJson(query);
+        String json = toJson(request);
         TDSavedQuery result =
                 doPost(
-                        buildUrl("/v3/schedule/create", query.getName()),
+                        buildUrl("/v3/schedule/create", request.getName()),
                         ImmutableMap.<String, String>of(),
                         Optional.of(json),
                         TDSavedQuery.class);
         return result;
     }
 
     @Override
-    public TDSavedQuery updateSavedQuery(String name, TDSavedQueryBuilder request)
+    public TDSavedQuery updateSavedQuery(String name, TDSavedQueryUpdateRequest request)
     {
         List<TDSavedQuery> savedQueries = listSavedQueries();
         for (TDSavedQuery q : savedQueries) {
             if (q.getName().equals(name)) {
+                /**
```

Comment:
I'm still not sure it's great to have this method in a public library until we can support it properly. Even if we don't provide this method now, users (and we) can implement it themselves and thus be aware of the race condition pitfall.

I'll let you decide.